src/distPlus/symbolic/SymbolicDist.re

```diff
@@ -296,103 +313,407 @@ module GenericSimple = {
-    shape |> E.O.toExt("");
+  let evaluateCombinationDistribution = (et1: nodeResult, et2: nodeResult, op: operation, n: int) => {
+    /* return either a Distribution or a RenderedShape. Must integrate to 1. */
+
+    let func = funcFromOp(op);
+    switch ((et1, et2, op)) {
+    /* Known cases: replace symbolic with symbolic distribution */
+    | (`Simple(`Float(v1)), `Simple(`Float(v2)), _) => {
+        `Simple(`Float(func(v1, v2)))
+      }
+
+    | (`Simple(`Normal(n2)), `Simple(`Float(v1)), `AddOperation)
+    | (`Simple(`Float(v1)), `Simple(`Normal(n2)), `AddOperation) => {
+        let n: normal = {mean: v1 +. n2.mean, stdev: n2.stdev};
+        `Simple(`Normal(n))
+      }
+
+    | (`Simple(`Normal(n1)), `Simple(`Normal(n2)), `AddOperation) => {
+        let n: normal = {mean: n1.mean +. n2.mean, stdev: sqrt(n1.stdev ** 2. +. n2.stdev ** 2.)};
+        `Simple(`Normal(n));
+      }
+
+    | (`Simple(`Normal(n1)), `Simple(`Normal(n2)), `SubtractOperation) => {
+        let n: normal = {mean: n1.mean -. n2.mean, stdev: sqrt(n1.stdev ** 2. +. n2.stdev ** 2.)};
+        `Simple(`Normal(n));
+      }
+
+    | (`Simple(`Lognormal(l1)), `Simple(`Lognormal(l2)), `MultiplyOperation) => {
+        let l: lognormal = {mu: l1.mu +. l2.mu, sigma: l1.sigma +. l2.sigma};
+        `Simple(`Lognormal(l));
+      }
+
+    | (`Simple(`Lognormal(l1)), `Simple(`Lognormal(l2)), `DivideOperation) => {
+        let l: lognormal = {mu: l1.mu -. l2.mu, sigma: l1.sigma +. l2.sigma};
+        `Simple(`Lognormal(l));
+      }
+
+
+    /* General cases: convolve the XYShapes */
+    | (`Simple(d1), `Simple(d2), _) => {
+        let (sc1, sd1) = renderDistributionToXYShape(d1, n);
+        let (sc2, sd2) = renderDistributionToXYShape(d2, n);
+        let (sc, sd) = combinationDistributionOfXYShapes(sc1, sd1, sc2, sd2, func);
+        `RenderedShape(sc, sd, 1.0)
+    }
+    | (`Simple(d1), `RenderedShape(sc2, sd2, i2), _)
+    | (`RenderedShape(sc2, sd2, i2), `Simple(d1), _) => {
+        let (sc1, sd1) = renderDistributionToXYShape(d1, n);
+        let (sc, sd) = combinationDistributionOfXYShapes(sc1, sd1, sc2, sd2, func);
+        `RenderedShape(sc, sd, i2)
+    }
+    | (`RenderedShape(sc1, sd1, i1), `RenderedShape(sc2, sd2, i2), _) => {
+        // sum of two multimodals that have a continuous and discrete each.
+        let (sc, sd) = combinationDistributionOfXYShapes(sc1, sd1, sc2, sd2, func);
+        `RenderedShape(sc, sd, i1);
+      }
+    }
   };
 
-  let toString = (dists: t) => {
-    let distString =
-      dists
-      |> E.A.fmap(d => GenericSimple.toString(fst(d)))
-      |> Js.Array.joinWith(",");
-    let weights =
-      dists
-      |> E.A.fmap(d =>
-           snd(d) |> Js.Float.toPrecisionWithPrecision(~digits=2)
-         )
-      |> Js.Array.joinWith(",");
-    {j|multimodal($distString, [$weights])|j};
+  let evaluatePointwiseSum = (et1: nodeResult, et2: nodeResult, n: int) => {
```

Comment:
what is ``et1`` stand for? It's not clear to me.

Reply:
`evaluatedTreeNode`. I'll rethink the whole naming scheme to make it clearer

Reply:
Another thing that could help is having things in nice modules. Then the "main" type inside could be called "t" (this is convention), and you can use "t1" and "t2" for var names, which is simple enough.

Reply:
That's what I'll do. Thanks!